br/pkg/task/restore.go

```diff
@@ -293,6 +297,37 @@ func (cfg *RestoreConfig) LocalEncryptionEnabled() bool {
 	return cfg.CipherInfo.CipherType != encryptionpb.EncryptionMethod_PLAINTEXT
 }
 
+type immutableRestoreConfig struct {
+	CmdName           string
+	UpstreamClusterID uint64
+	Storage           string
+	ExplictFilter     bool
+	FilterStr         []string
+	WithSysTable      bool
+}
+
+func Hash(cmdName string, cfg *RestoreConfig) ([]byte, error) {
+	if cfg == nil {
+		return nil, errors.New("nil config")
+	}
+
+	config := immutableRestoreConfig{
+		CmdName:           cmdName,
+		UpstreamClusterID: cfg.UpstreamClusterID,
+		Storage:           cfg.Storage,
+		ExplictFilter:     cfg.ExplicitFilter,
```

Comment:
like we discussed in the chat before, if a restore with default filter A failed and later user restore with exact same filter A but explicit, all other fields remain the same, I feel like we could reuse the checkpoint, but having explicit filter in the hash will prevent from doing so